pkg/api/org_invite.go

```diff
@@ -91,25 +92,37 @@ func (hs *HTTPServer) AddOrgInvite(c *contextmodel.ReqContext) response.Response
 	}
 
 	if hs.Cfg.DisableLoginForm {
-		return response.Error(400, "Cannot invite external user when login is disabled.", nil)
+		return response.Error(http.StatusBadRequest, "Cannot invite external user when login is disabled.", nil)
 	}
 
 	cmd := tempuser.CreateTempUserCommand{}
-	cmd.OrgID = c.OrgID
+	cmd.OrgID = c.SignedInUser.GetOrgID()
 	cmd.Email = inviteDto.LoginOrEmail
 	cmd.Name = inviteDto.Name
 	cmd.Status = tempuser.TmpUserInvitePending
-	cmd.InvitedByUserID = c.UserID
+
+	namespace, identifier := c.SignedInUser.GetNamespacedID()
+	var userID int64
+	switch namespace {
+	case identity.NamespaceUser, identity.NamespaceServiceAccount:
+		var err error
+		userID, err = strconv.ParseInt(identifier, 10, 64)
+		if err != nil {
+			return response.Error(http.StatusInternalServerError, "Unrecognized user", err)
+		}
+	}
```

Comment:
thought: seems like logic similar to this will be repeated in several places. Do you think it's worth encapsulating it somewhere?